### PR TITLE
Do not interleave code and declarations in C files

### PR DESCRIPTION
**Pull request by tbeu**
Avoid C99 style to interleave code and declarations in order to compile msgpackc with MSVC

### CODE REVIEW

**Comment by Pyrohh** [comment]
Is this for an older MSVC? I could have sworn that MSVC 2013 supported C99 style declarations.

**Comment by tbeu** [comment]
I use VS2010.

**Comment by tbeu** [comment]
Just checked with simple.c example. It compiles with VS2013 but does not with VS2010 or VS2012.

**Comment by redboltz** [comment]
Hi @tbeu, thank you for sending the PR. It's worth supporting pre-C99 C (ansi-C). I will merge your PR. Before merging, I have some requests. I will comment to PR directly.

File: include/msgpack/sysdep.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ typedef unsigned __int64 uint64_t;
 #include <stdbool.h>
 #endif
 
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && !defined(_LIB)
```
Comment (redboltz) [inline include/msgpack/sysdep.h]: This is reasonable change to support static-lib on MSVC. However, it is a different topic. Could you separate it from the pull request?

File: include/msgpack/unpack_template.h [hunk 2]
```diff
@@ -93,10 +92,9 @@ msgpack_unpack_func(msgpack_unpack_object, _data)(msgpack_unpack_struct(_context
     return (ctx)->stack[0].obj;
 }
 
-
 msgpack_unpack_func(int, _execute)(msgpack_unpack_struct(_context)* ctx, const char* data, size_t len, size_t* off)
 {
-    assert(len >= *off);
+    /* assert(len >= *off); */
```
Comment (redboltz) [inline include/msgpack/unpack_template.h]: Commenting out the assertion changes a meaning of code in DEBUG mode.
Could you keep the assertion above like as follows?

```C++
    assert(len >= *off);
    {
        ...
```

File: include/msgpack/sysdep.h [hunk 3]
```diff
@@ -93,7 +93,6 @@ typedef unsigned int _msgpack_atomic_counter_t;
 #endif
 #endif
 
-
```
Comment (redboltz) [inline include/msgpack/sysdep.h]: The PR removes many blank lines. However, it does not relate to the fix. Could you keep the blank lines?

**Comment by tbeu** [comment]
OK, I can prepare a new PR. I will also consider the C examples.

Should I close this issue for now?

**Comment by redboltz** [comment]
Thank you very much! When you send the new PR, close this issue please.

**Comment by redboltz** [comment]
Thank you for refining your PR. I've just merged your new PR. Could you send the rest part of your original PR?
I mean 
```C++
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && !defined(_LIB)
```

**Pull request closed without merging**